docs/FRONTEND.md

```diff
@@ -0,0 +1,203 @@
+
+```
+.tweet { /* … */ }
+.tweet.is-expanded { /* … */ }
+```
+
+```
+<article class="tweet is-expanded">
+  …
+</article>
+```
+
+<a name="bem"></a>
+## BEM Recommendations
+
+### Keep selector level to minimum
+
+Heavily nested BEM selectors (e.g., `.event__sidebar__tickets__content`) are a sign you need to break a component out into smaller pieces.
+
+**Rule-of-thumb**: Three or more levels deep and you should be thinking about splitting up components.
+
+### Unintended modifiers
+
+Try to avoid modifiers that don’t have a root class e.g., if you have `list--stacked`, `list--inline` you must also have a `list` class. If you don’t then these are two separate components with a common name `list-stacked` and `list-inline`.
+
+### Avoid modifiers on children
+
+Modifiers on children can feel correct in BEM notation, after all it keeps the specificity low (one level deep), but still allows modification of child elements. However it can often cause a conceptual disconnect, as it’s not immediately clear where the modifier is intended to be used.
+
+Instead try to keep modifiers to the root level selector. For example given the following component:
+
+```
+.mod { color: hotpink }
+.mod__child { padding: 2%; }
+
+Instead of
+.mod__child--slim { padding: 1% }
+
+Prefer:
+
+.mod--slim {
+   .mod__child {
+      padding: 1%;
+   }
+}
+```
+
+While the former keeps specificity as low as possible, it’s easier to reason about the second level as we are documenting a known set of variants for a component. If you are consistently adding modifiers to child elements it’s a good indicator that the component is too large; that element should probably be the root of a new component.
+
+### Don’t use BEM children out of context.
+
+One risk with BEM naming conventions is that it’s possible to use a child-selector outside of the context of it’s original parent. So if you see something like the following then it’s a sign that your component wasn’t granular enough or you need to refactor your views:
+
+```
+<article class="some-component">
+	<h2 class="some-component__title"">Title<h2>
+  <p class="urelated-component__child"></p>
+</article>
+```
+
+<a name="javascript"></a>
+## JavaScript
+
+> Separate style and behavior concerns by using `.js-` prefixed classes for behavior.
+
+**`.js-` classes should never appear in your stylesheets.** They are for JavaScript only. Inversely, there is rarely a reason to see presentation classes like `.header-nav-button` in JavaScript. You should use state classes like `.is-state` in your JavaScript and your stylesheets as `.component.is-state`.
+
+<a name="prefixes"></a>
+## Vendor Prefixes
+
+We use [Autoprefixer](https://github.com/postcss/autoprefixer) for vendor prefixes so write any prefixed CSS using the spec version only and the build process will add any requried prefixes. It's suprising how few vendor prefixes a anymore.
```

Comment:
"It's suprising how few vendor prefixes a anymore"

I think you accidentally a word.